app/models/event.rb

```diff
@@ -10,4 +11,8 @@ class Event < ApplicationRecord
             presence: true
 
   scope :upcoming, -> { where('date >= ?', Time.zone.today) }
+  scope :requiring_notification, -> do
+    where('date = ?', Date.current - NOTIFICATION_PERIOD)
+      .includes(presentations: [:person])
```

Comment:
Tried this in console and it grabs only presentations that have a person as well as loading the people in memory.